droidlet/dialog/dialogue_task.py

```diff
@@ -0,0 +1,206 @@
+        self.asked = False
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    @Task.step_wrapper
+    def step(self):
+        """Ask a confirmation question and wait for response."""
+        # Step 1: ask the question
+        if not self.asked:
+            task_list = [
+                Say(self.agent, {"response_options": self.question}),
+                AwaitResponse(self.agent, {"asker_memid": self.memid}),
+            ]
+            self.add_child_task(maybe_task_list_to_control_block(task_list, self.agent))
+            self.asked = True
+            return
+        # Step 2: check the response and add the task if necessary
+        self.finished = True
+        # FIXME: change this to sqly when syntax for obj searches is settled:
+        t = self.agent.memory.get_triples(subj=self.memid, pred_text="dialogue_task_response")
+        chat_mems = [self.agent.memory.get_mem_by_id(triples[2]) for triples in t]
+        if not chat_mems:
+            return
+        if any([c.chat_text in MAP_YES for c in chat_mems]):
+            for m in self.task_memids:
+                # activate
+                TaskNode(self.agent.memory, m).get_update_status({"prio": 1, "paused": 0})
+        else:
+            for m in self.task_memids:
+                # mark tasks as finished
+                TaskNode(self.agent.memory, m).get_update_status({"prio": -2})
+        return
+
+
+class ConfirmReferenceObject(Task):
+    """This Task confirms if a reference object is correct.
+
+    if there is a response that maps to "yes"  it will place a triple
+    (subj=self.memid, pred_text="dialogue_task_output", obj_text="yes")
+    else (either no response, or maps no, or otherwise)
+    (subj=self.memid, pred_text="dialogue_task_output", obj_text="no")
+
+    Args:
+        bounds: general area of reference object to point at
+        pointed: flag determining whether the agent pointed at the area
+        asked: flag determining whether the confirmation was asked for
+
+    """
+
+    def __init__(self, agent, task_data={}):
+        task_data["blocking"] = True
+        super().__init__(agent, task_data=task_data)
+        r = task_data.get("reference_object")
+        if hasattr(r, "get_point_at_target"):
+            self.bounds = r.get_point_at_target()
+        else:
+            # this should be an error
+            self.bounds = tuple(np.min(r, axis=0)) + tuple(np.max(r, axis=0))
+        self.pointed = False
+        self.asked = False
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    @Task.step_wrapper
+    def step(self):
+        """Confirm the block object by pointing and wait for answer."""
+        if not self.asked:
+            task_list = [
+                Say(self.agent, {"response_options": self.question}),
+                AwaitResponse(self.agent, {"asker_memid": self.memid}),
+            ]
+            self.add_child_task(task_list)
```

Comment:
nit : we should perhaps rename `add_child_task` to `add_children_tasks` if it can take in a list